pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1610,9 +1610,13 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
                                   PyArray_DescrFromType(NPY_DOUBLE));
         tc->type = JT_DOUBLE;
         return;
-    } else if (PyArray_Check(obj) && PyArray_CheckScalar(obj)) {
+    } else if (PyArray_CheckScalar(obj)) {
+        /* This handles all cases of array of zero dimension (numpy.array(1)) OR
```

Comment:
You can remove the comment - I think this is inferrable already from the code/tests

Reply:
Hi @WillAyd, I have incorporated this change in latest commit. Thanks for this suggestion.